ceno_zkvm/src/scheme/prover.rs

```diff
@@ -183,9 +183,10 @@ impl<
 
                 let circuit_idx = self.pk.circuit_name_to_index.get(name).unwrap();
                 // write (circuit_idx, num_var) to transcript
-                transcript.append_message(&circuit_idx.to_le_bytes());
+                transcript.append_field_element(&E::BaseField::from_canonical_usize(*circuit_idx));
```

Comment:
The behavior of `usize.to_le_bytes()` is different on 64-bit cpu and 32-bit cpu. This kind of divergence is not desirable.